### PR TITLE
New functions

### PR DESCRIPTION
**Pull request by Jackojc**
- Utility function for conditionally running a function argument with a compile time flag.
- Utility functions for directly printing repeated characters. This means you can do `print_tabs(1)` instead of `print(tabs(1))`
- Emphasised variants of logging which embolden the first argument but leave all others as normal. Handy for key:value printing.
- Heading variants of logging functions which embolden the entire list of arguments passed in. Less of `print(style::bold, "Hello")`.
- Added compile time variants of every logging function so you can conditionally choose whether to compile things like debug printing depending on a flag's value.

### CODE REVIEW

File: module.h [hunk 1]
```diff
@@ -105,6 +105,16 @@ namespace ws::module {
 
 
 
+    // Wrappers to help implement compile time checks for compiling code.
+    template <bool B, typename F>
+    constexpr void run_if(const F& func) {
```
Comment (Hazurl) [inline module.h]: Should use a universal reference `F&&` and `std::forward<F>(func)()`

**Comment by Hazurl** [comment]
Thx a lot for cleaning the history 👌

**Comment by Jackojc** [comment]
Good idea with the `WS_MODULE_LOG_OFF`, I'll add it.

**Comment by Jackojc** [comment]
How's that @Hazurl ?

File: module.h [hunk 2]
```diff
@@ -328,19 +347,31 @@ namespace ws::module {
...
+        #else
+            return ws::module::printer;
+
+        #endif
     }
 
 
 
     template <typename... Ts>
     inline std::ostream& pipe(Ts&&... args) {
```
Comment (Hazurl) [inline module.h]: only the log, not the piping...

Comment (Jackojc) [reply module.h]: I'll add one for pipes too: `WS_MODULE_PIPE_OFF`

**Comment by Jackojc** [comment]
Ready to merge if approved.

**Comment by Hazurl** [comment]
Gotta resolved the conflict

**Pull request merged**